### DISCUSSION TITLE
Using Avalonia with AppCenter

### DISCUSSION THREAD

**Discussion by saigkill** [AvaloniaUI/Avalonia#13927] [Q&A] [answered]
I would to write an App what uses the MS AppCenter. 

As far as i understand, the official AppCenter SDK doesn't provides Avalonia. So i found that mod https://www.nuget.org/packages/Aigio.Any.AppCenter what works as the official one.

As the docs saying i have to add the Start Method to a overrided OnStartup method.

As far is i understand, the Avalonia Application Baseclass doesn't have such method.

What would be the equivalent on Avalonia?

**Comment by jp2masa** [comment]
Probably `Application.OnFrameworkInitializationCompleted`.

**Comment by maxkatz6** [comment, marked as answer]
For the most part, AppCenter should stop hardcoding its SDKs to the specific frameworks or platforms.
But you still can achieve AppCenter working with some minimal changes.
I did it before in my fork https://github.com/maxkatz6/appcenter-sdk-dotnet 
https://www.nuget.org/packages/AppCenter.Standard/
https://www.nuget.org/packages/AppCenter.Crashes.Standard/
https://www.nuget.org/packages/AppCenter.Analytics.Standard/

Note - these packages can be used with any framework or any platform that supports .NET. Assuming SQLite works there.

**Reply by maxkatz6** [reply]
Also, I am not going to maintain these packages. But feel free to fork and modify anything, as it's under MIT license.
